src/Middleware.php

```diff
@@ -3,31 +3,127 @@
 namespace Inertia;
 
 use Closure;
-use Illuminate\Support\Facades\Response;
-use Symfony\Component\HttpFoundation\RedirectResponse as Redirect;
+use Illuminate\Http\Request;
+use Illuminate\Support\Collection;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
 
-class Middleware
+abstract class Middleware
 {
-    public function handle($request, Closure $next)
+    /**
+     * Determines the current Inertia asset version hash.
```

Comment:
I would just say:

```
Determines the current asset version.
```

Since *technically* it doesn't have to be a hash (even if we end up having a default implementation that does do this).